runtime/src/bank.rs

```diff
@@ -5241,18 +5266,16 @@ impl Bank {
 
 impl Drop for Bank {
     fn drop(&mut self) {
-        if self.skip_drop.load(Relaxed) {
-            return;
-        }
-
```

Comment:
We remove this because now even after a call to `remove_unrooted_slot()` on a bank, we still want to run `purge_slot` to remove the entry for the bank from the newly introduced `AccountsIndex::removed_slot_ids`